cpp/include/cudf/transform.hpp

```diff
@@ -140,6 +140,31 @@ std::pair<std::unique_ptr<cudf::table>, std::unique_ptr<cudf::column>> encode(
   cudf::table_view const& input,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
+/**
+ * @brief Encode the categories in column with an indicator function
```

Comment:
I think a reader would wonder "what's an indicator function?"
```suggestion
 * @brief Encodes `input` by generating a new column for each value in `categories` indicating the presence of that value in `input`. 
 *
 * The resulting per-category columns are returned concatenated as a single column viewed by a `table_view`.
 *
 * The `i`th row of the `j`th column in the output table equals 1
 * if `input[i] == categories[j]`, and 0 otherwise.
```